src/main/java/modbus/SlaveFolder.java

```diff
@@ -427,8 +430,7 @@ protected void readPoint(Node pointNode) {
 			return;
 		}
 
-		if (!ModbusConnection.NODE_STATUS_CONNECTED.equals(conn.statnode.getValue().getString())) {
-			conn.checkConnection();
+		if (!NODE_STATUS_READY.equals(root.getStatusNode().getValue().getString()) && !checkConnectionByDevice()) {

```

Comment:
I'd rename `checkConnectionByDevice()` for `isDeviceConnected` or something more "query"-y